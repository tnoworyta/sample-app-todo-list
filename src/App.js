import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux';
import { addTodo, removeTodo } from './store'

export class TodoAdd extends Component {
  state = {
  }
  updateValue = ({ target: { value } }) => this.setState({ value });
  addTodo = () =>  {
    this.props.addTodo(this.state.value);
    this.setState({ value: '' });
  }

  render() {
    return (
      <div>
        <input value={this.state.value} onChange={this.updateValue}/>
        <button onClick={this.addTodo}></button>
      </div>
    );
  }
}

export class TodoList extends Component {
  removeTodo = (todo) => () => this.props.removeTodo(todo);

  render() {
    return (
      <ul>
        {this.props.todos.map(todo => <li key={todo.id}>{todo.text} <button onClick={this.removeTodo(todo)}></button></li>)}
      </ul>
    );
  }
}

class App extends Component {


  addTodo = (text) => this.props.addTodo(text);

  removeTodo = (todo) => this.props.removeTodo(todo);

  render() {
    const { todos } = this.props;
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        <div className="App-intro">
          <TodoAdd addTodo={this.addTodo}/>
          <TodoList todos={todos} removeTodo={this.removeTodo}/>
          <div>Count: {todos.length}</div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = ({ todos }) => ({
  todos
});

const mapDispatchToProps = (dispatch) => bindActionCreators({
  addTodo,
  removeTodo,
}, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(App);
