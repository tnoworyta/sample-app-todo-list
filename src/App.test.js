import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App, { TodoAdd } from './App';
import { Provider } from 'react-redux'

test('render App', () => {
  const store = { }
  const wrapper = shallow(<Provider><App/></Provider>);
  expect(wrapper).toMatchSnapshot();
});

test('render TodoAdd', () => {
  const wrapper = shallow(<TodoAdd />);

  expect(wrapper).toMatchSnapshot();
});

test('render TodoAdd', () => {
  const wrapper = shallow(<TodoAdd />);

  expect(wrapper).toMatchSnapshot();
});
