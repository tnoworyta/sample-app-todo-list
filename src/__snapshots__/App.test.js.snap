// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App 1`] = `<Connect(App) />`;

exports[`render TodoAdd 1`] = `
<div>
  <input
    onChange={[Function]}
  />
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`render TodoAdd 2`] = `
<div>
  <input
    onChange={[Function]}
  />
  <button
    onClick={[Function]}
  />
</div>
`;
